sys/net/application_layer/uhcp/uhcpc.c

```diff
@@ -31,44 +23,32 @@ static void _timeout(void *arg) {
  */
 void uhcp_client(uhcp_iface_t iface)
 {
-    ipv6_addr_t target;
-    ipv6_addr_from_str(&target, "ff15::abcd");
+    sock_udp_t sock;
+    sock_udp_ep_t local = { .family=AF_INET6, .port=UHCP_PORT, .netif=iface };
+    sock_udp_ep_t req_target = { .family=AF_INET6, .port=UHCP_PORT, .netif=iface };
+    sock_udp_ep_t remote;
+
+    inet_pton(AF_INET6, "ff15::abcd", req_target.addr.ipv6);
 
     /* prepare UHCP header */
     uhcp_req_t req;
     uhcp_hdr_set(&req.hdr, UHCP_REQ);
     req.prefix_len = 64;
 
     /* create listening socket */
-    ipv6_addr_t zero = {{0}};
-    conn_udp_t conn;
-    int res = conn_udp_create(&conn, &zero, 16, AF_INET6, UHCP_PORT);
+    int res = sock_udp_create(&sock, &local, NULL, 0);
 
-    uint8_t srv_addr[16];
-    size_t srv_addr_len;
-    uint16_t srv_port;
     uint8_t buf[sizeof(uhcp_push_t) + 16];
 
-    kernel_pid_t pid = thread_getpid();
-    xtimer_t timeout;
-    timeout.callback = _timeout;
-    timeout.arg = &pid;
-
     while(1) {
-        xtimer_set(&timeout, 10U*SEC_IN_USEC);
         puts("uhcp_client(): sending REQ...");
-        conn_udp_sendto(&req, sizeof(uhcp_req_t), NULL, 0, &target, 16, AF_INET6 , 12345, 12345);
-        res = conn_udp_recvfrom(&conn, buf, sizeof(buf), srv_addr, &srv_addr_len, &srv_port);
+        sock_udp_send(&sock, &req, sizeof(uhcp_req_t), &req_target);
+        res = sock_udp_recv(&sock, buf, sizeof(buf), 10U*SEC_IN_USEC, &remote);
         if (res > 0) {
-            xtimer_remove(&timeout);
-            uhcp_handle_udp(buf, res, srv_addr, srv_port, iface);
+            uhcp_handle_udp(buf, res, remote.addr.ipv6, remote.port, iface);
             xtimer_sleep(60);
         }
         else {
```

Comment:
`res < 0` can also be any other reception error, so the error message below might be misleading. Either make the error message more generic or introduce a specific case for timeout checking for `-ETIMEDOUT`